Fixed documentation bugs introduced in PR#641

emilotep opened the pull request:
## Change Summary
Removed extra newline inadvertently added while rendering BGP neighbor documentation. Bug introduced in PR#641.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)
None.

## Component(s) name

`arista.avd.eos_cli_config_gen

## Proposed changes
Removed newline at line 95 in router-bgp.j2 documentation template.

## How to test
Ran molecule to ensure extra linebreak is gone.

## Checklist:
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

emilotep committed:
Fixed extra linebreak in documentation added in PR#641

emilotep committed:
Fixed another BGP Neighbors documentation bug

The bug caused BGP Neighbors documentation table to lack
header if no default VRF neighbors are defined, appeared
in router-bgp-vrf-lite.md document.

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/router-bgp.j2:
```diff
@@ -82,6 +82,13 @@
 {% if router_bgp.vrfs is arista.avd.defined %}
 {%     for vrf in router_bgp.vrfs | arista.avd.natural_sort %}
 {%         if router_bgp.vrfs[vrf].neighbors is arista.avd.defined %}
+{%             if router_bgp.neighbors is not arista.avd.defined %}
```
It would be more elegant to add a check for vrf neighbors in the section above in line 66, to avoid repeating headers like this. This is very confusing to read.
So just do a for loop for bgp_neighbors and another for loop for the vrf neighbors.

emilotep replied:
Fixed in commit 00c400b.

ClausHolbechArista reviewed (changes requested):
Please refactor the condition for the case where you only have vrf neighbors.

emilotep committed:
Improved documentation templating

emilotep committed:
Fixed missing newline at EOF on some roles tasks files

This was making the pre-commit fail. I did not change these files in my
fork at all so I don't know why this failure appears now.

ClausHolbechArista committed:
Update CI Artifacts

ClausHolbechArista committed:
Update main.yml

Removing old playbook remains from another PR. Maybe you forgot to rebase?

ClausHolbechArista reviewed (approved):
LGTM!

I have updated all molecule CI artifacts and fixed some rebase issue, so it is ready to merge.
